pkg/local_object_storage/blobstor/fstree/fstree_write_linux.go

```diff
@@ -4,26 +4,47 @@ package fstree
 
 import (
 	"errors"
+	"io/fs"
 	"strconv"
 
 	"github.com/nspcc-dev/neofs-node/pkg/local_object_storage/blobstor/common"
 	"golang.org/x/sys/unix"
 )
 
-func (t *FSTree) writeData(p string, data []byte) error {
-	err := t.writeFile(p, data)
+type linuxWriter struct {
+	root  string
+	perm  uint32
+	flags int
+}
+
+func newSpecificWriteData(root string, perm fs.FileMode, noSync bool) func(string, []byte) error {
+	flags := unix.O_WRONLY | unix.O_TMPFILE | unix.O_CLOEXEC
+	if !noSync {
+		flags |= unix.O_DSYNC
+	}
+	fd, err := unix.Open(root, flags, uint32(perm))
+	if err != nil {
+		return nil
```

Comment:
can we comment this key difference b/w the old and the new behaviors? it took me some time to understand how exactly this PR fixes #2609

Reply:
Four cited lines look wonderful on their own, I agree.